### PR TITLE
Use npm module 'is-local-host' to determine if a URL is a local host or not

### LINKED ISSUE #9748: Generalize how we detect localhost
Do we need something more general here? Like so for edge cases? https://www.npmjs.com/package/is-localhost-ip

_Originally posted by @IanMatthewHuff in https://github.com/microsoft/vscode-jupyter/pull/9745#discussion_r856365596_

### PR DESCRIPTION
**Pull request by rchiodo**
Fixes #9748

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2170,6 +2170,7 @@
         "iconv-lite": "^0.4.21",
         "image-size": "^1.0.0",
         "inversify": "^5.0.1",
+        "is-localhost-ip": "^1.4.0",
```
Comment (IanMatthewHuff) [inline package.json:2173]: The npmjs entry actually said GPL, but from the actual repro it looks like it was licensed to MIT years ago. I'd assume that's ok? https://github.com/tinovyatkin/is-localhost-ip/commit/f9172384513ddca3245c460dbf66ef5234e8311c

Comment (IanMatthewHuff) [reply package.json:2173]: Wondering if component governance uses npm entry...

Comment (rchiodo) [reply package.json:2173]: I guess we'll find out :)

**Comment by rchiodo** [comment]
Blech I can't find an npm module that doesn't have node dependencies. I'm just going to close this for now.

**Pull request closed without merging**